Lib/profiling/sampling/gecko_collector.py

```diff
@@ -0,0 +1,451 @@
+import json
+import os
+import platform
+import sys
+import time
+from typing import Dict, List, Optional, Tuple, Any
+
+from .collector import Collector, THREAD_STATE_RUNNING
+
+
+class GeckoCollector(Collector):
+    def __init__(self, *, skip_idle=False):
+        self.skip_idle = skip_idle
+        self.start_time = time.time() * 1000  # milliseconds since epoch
+
+        # Global string table (shared across all threads)
+        self.global_strings = ["(root)"]  # Start with root
```

Comment:
ℹ️ **Maintainability**

Consider using a constant for the root string to improve maintainability and readability. This avoids magic strings scattered throughout the code.

**Suggested fix:**
```suggestion
        ROOT_STRING = "(root)"
        self.global_strings = [ROOT_STRING]
```